src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +284,165 @@ def get_info(
         self, request: InfoRequest, context: grpc.ServicerContext
     ) -> InfoResponse:
         return InfoResponse(**info_response_kv_args())
+
+    def _execute_task_in_thread(
+        self, run_request: RunTaskRequest, task_execution: _TaskExecution
+    ):
+        response = RunTaskResponse(
+            task_id=run_request.task_id,
+            outcome_code=TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE,
+            failure_reason=TaskFailureReason.TASK_FAILURE_REASON_INTERNAL_ERROR,
+            stderr="",
+            stdout="",
+        )
+
+        try:
+
+            def progress_reporter(progress: Progress):
+                with task_execution.updated:
+                    task_execution.progress = ProgressUpdate(
+                        current=progress.current, total=progress.total
+                    )
+                    task_execution.updated.notify_all()
+                # sleep(0) here momentarily releases the GIL, giving
+                # other threads a chance to run.
+                time.sleep(0)
```

Comment:
Let's not keep time.sleep it in the code base unless we have a very clear reason why, otherwise after years ppl would be worried to remove it cause they don't know what's going to happen.